Add pod annotations (#1849)
googleforgames/agones: site/content/en/docs/Installation/Install Agones/helm.md

| `agones.controller.nodeSelector`                    | Controller [node labels][nodeSelector] for pod assignment                                       | `{}`                   |
| `agones.controller.tolerations`                     | Controller [toleration][toleration] labels for pod assignment                                   | `[]`                   |
| `agones.controller.affinity`                        | Controller [affinity][affinity] settings for pod assignment                                     | `{}`                   |

> | `agones.controller.annotations`                     | Controller [annotations][annotations] settings for pod assignment                               | `{}`                   |

Reviewer: Please move this down to where we have the section "New Configuration Features:" - so they only get displayed on the next release :+1: 

Context: https://agones.dev/site/docs/contribute/#within-a-page

Author: Thank you. I fixed it with ab93fd6b5e51ebf65ce76e132a068b1e5584ff2e.

| `agones.allocator.disableTLS`                       | Turns off TLS security for incoming connections to the allocator. | `false`                |
| `agones.allocator.tolerations`                      | Allocator [toleration][toleration] labels for pod assignment                                    | `[]`                   |
| `agones.allocator.affinity`                         | Allocator [affinity][affinity] settings for pod assignment                                      | `{}`                   |

> | `agones.allocator.annotations`                      | Allocator [annotations][annotations] settings for pod assignment                                | `{}`                   |

Reviewer: Please also move this down to where we have the section "New Configuration Features:" - so they only get displayed on the next release :+1:

Author: Thank you. I fixed it with ab93fd6b5e51ebf65ce76e132a068b1e5584ff2e.

| Parameter                                           | Description                                                                                     | Default                |
| --------------------------------------------------- | ----------------------------------------------------------------------------------------------- | ---------------------- |

> | `agones.controller.annotations`                     | Controller [annotations][annotations] settings for pod assignment                               | `{}`                   |

Reviewer: ```suggestion
| `agones.controller.annotations`                     | [Annotations][annotations] added to the Agones controller pods                               | `{}`                   |
```

Just going by Robbie's suggestion, to make it super clear. WDYT?

Author: Fixed.

| `agones.allocator.annotations`                      | Allocator [annotations][annotations] settings for pod assignment                                | `{}`                   |
| `agones.ping.annotations`                           | Ping [annotations][annotations] settings for pod assignment                                     | `{}`                   |

> | `agones.ping.http.annotations`                      | [Annotations](annotations) for ping http service                                                | `{}`                   |

Author: `[Annotations](annotations)` 😭

> | `agones.ping.http.annotations`                      | [Annotations][annotations] for ping http service                                                | `{}`                   |

Reviewer: Sorry to make you juggle this around, but `agones.ping.udp.annotations` and `agones.ping.http.annotations` already exist in the current release, but aren't documented - so they can go in the section up above.

There new, remaining values should stay down here.

Random thought - do we think there is any confusion given that `agones.ping.udp.annotations` is a annotation on a service, but `agones.controller.annotations` is on Pods?

Author: > Random thought - do we think there is any confusion given that agones.ping.udp.annotations is a annotation on a service, but agones.controller.annotations is on Pods?

Hmmm, it's somewhat confusing.
Should I revise it to `agones.controller.podAnnotations` or `agones.ping.udp.serviceAnnotations` or something?

Author: > Sorry to make you juggle this around, but agones.ping.udp.annotations and agones.ping.http.annotations already exist in the current release, but aren't documented - so they can go in the section up above.
>
> There new, remaining values should stay down here.

I understand.
I'll fix the name I asked for in my comment above when I decide what to do with it.

Reviewer: @roberthbailey @pooneh-m would love a second opinion?

Worth noting, `agones.ping.udp.annotations` and `agones.ping.http.annotations` where in previous releases, but weren't documented.

WDYT?

Reviewer 2: If you look at the structure in https://github.com/googleforgames/agones/blob/master/install/helm/agones/values.yaml I don't actually think it's all that confusing. The annotations for the service / pod are clearly nested under the values for the service / pod respectively. 

I think the descriptions in this table should make it clear what type of k8s resource the annotations are being applied to. e.g.

| `agones.controller.annotations` | [Annotations][annotations] added to the Agones controller pods |
| `agones.ping.http.annotations ` | [Annotations][annotations] added to the Agones ping http service |

Reviewer: SGTM! Thanks for the second/third pair of eyes - that works for me :+1:

Author: Fixed :+1:

> | `agones.ping.http.annotations`                      | [Annotations][annotations] for ping http service                                                | `{}`                   |

Reviewer: ```suggestion
| `agones.ping.http.annotations`                      | [Annotations][annotations] added to the Agones ping http service                                             | `{}`                   |
```

Minor tweaks for just some extra clarity based on Robbie's suggestions. Just to make things extra clear.

Author: Fixed.
